src/main/java/org/sunbird/profile/service/ProfileServiceImpl.java

```diff
@@ -338,6 +350,150 @@ public SBApiResponse profileUpdate(Map<String, Object> request, String userToken
+						missingAttribts.add(field);
+					}
+				}
+				if (!cadreMap.containsKey(Constants.CADRE_BATCH)
+						|| cadreMap.get(Constants.CADRE_BATCH) == null ||
+						!(cadreMap.get(Constants.CADRE_BATCH) instanceof Integer)) {
+					missingAttribts.add(Constants.CADRE_BATCH);
+				}
+
+				if (missingAttribts.size() > 0) {
+					return "Missing mandatory attributes or Improper dataType. " + missingAttribts.toString();
+				}
+
+				Map<String, Object> propertyMap = new HashMap<>();
+				propertyMap.put(Constants.ID, Constants.CADRE_CONFIG);
+				cadreSchemaDetails = cassandraOperation.getRecordsByPropertiesWithoutFiltering(
+						Constants.KEYSPACE_SUNBIRD, Constants.TABLE_SYSTEM_SETTINGS, propertyMap,
+						null, 2);
+				if (cadreSchemaDetails.isEmpty() && !cadreSchemaDetails.contains(Constants.VALUE)) {
+					return "Failed to validate cadreDetails schema";
+				}
+				boolean isCivilTypePresent = false;
+				boolean isServiceTypePresent = false;
+				boolean isCadrePresent = false;
+
+				for (Map<String, Object> cadreDetail : cadreSchemaDetails) {
+					String mapString = (String) cadreDetail.get(Constants.VALUE);
+					Map<String, Object> map = mapper.readValue(mapString,
+							new TypeReference<Map<String, Object>>() {
+							});
+
+					Map<String, Object> civilServiceMap = (Map<String, Object>) map.get(
+							Constants.CIVIL_SERVICE_TYPE);
+					List<Map<String, Object>> civilServiceList = (List<Map<String, Object>>) civilServiceMap.get(
+							"civilServiceTypeList");
+
+					for (Map<String, Object> eachCivilService : civilServiceList) {
+						String civilServiceName = (String) eachCivilService.get(Constants.NAME);
+						if (((String) cadreMap.get(Constants.CIVIL_SERVICE_TYPE)).equalsIgnoreCase(
+								civilServiceName)) {
+							isCivilTypePresent = true;
+							List<Map<String, Object>> serviceMap = (List<Map<String, Object>>) eachCivilService.get(
+									Constants.SERVICE_TYPE);
+
+							for (Map<String, Object> service : serviceMap) {
+								String serviceTypeName = (String) service.get(Constants.NAME);
+								if (((String) cadreMap.get(Constants.CIVIL_SERVICE_NAME)).equalsIgnoreCase(
+										serviceTypeName)) {
+									isServiceTypePresent = true;
+									List<Map<String, Object>> cadreList = (List<Map<String, Object>>) service.get(
+											Constants.CADRE_LIST);
+
+									isCadrePresent = cadreList.stream()
+											.anyMatch(eachCadre ->
+													((String) cadreMap.get(Constants.CADRE_NAME)).equalsIgnoreCase(
+															(String) eachCadre.get(Constants.NAME))
+											);
+								}
+							}
+						}
+					}
+				}
+				if (!isServiceTypePresent) {
+					return "Invalid civilserviceType is given";
+				}
+				if (!isCivilTypePresent) {
+					return "Invalid civilSerivceName is given";
+				}
+				if (!isCadrePresent) {
+					return "Invalid cadreName is given";
+				}
+
+//				for (Map cadreDetail : cadreSchemaDetails) {
```

Comment:
Remove the code instead of commenting out

Reply:
done